src/auth/googleauth.ts

```diff
@@ -634,4 +643,59 @@ export class GoogleAuth {
           if (callback) callback(null, body);
         });
   }
+
+
+  /**
+   * Returns the contents of the metadata of GC instance
+   * @return object representation of the service account JSON
+   */
+  getCredentials(
+      callback?: (err: Error|null, credentials?: CredentialBody) => void) {
+    if (this.jsonContent) {
+      const credential: CredentialBody = {
+        client_email: this.jsonContent.client_email,
+        private_key: this.jsonContent.private_key
+      };
+      if (callback) {
+        callback(null, credential);
+      }
+    } else {
+      this._checkIsGCE((err, gce) => {
+        if (err) {
+          if (callback) {
+            callback(err, undefined);
+          }
+        } else if (gce) {
+          // For GCE, return the service account details from the metadata
+          // server
+          this.transporter.request(
+              {
+                method: 'GET',
+                uri:
+                    'http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/?recursive=true',
+                headers: {'Metadata-Flavor': 'Google'}
+              },
+              (err, body, res) => {
+                if (err || !res || res.statusCode !== 200 || !body ||
+                    !body.default || !body.default.email) {
```

Comment:
Please make sure the tests are covering these paths though.

Reply:
Done.